ansible/awx
# Hide survey passwords in saved launch configs

This is some follow-up work to #169, but the UI impact should be minimal.

The main change is that we will hide answers to survey questions - even if those answers are provided on a saved launch configuration.

The previous PR was already attempting to do this, but it missed the critical step of actually saving the password list on the creation or modification of the configuration. Otherwise, there is legwork done to combine both the variables from the node, and the password list from the node.

Tested:

 - Create a JT with password question survey
 - Create a WFJT, with no extra_vars
 - Add a node that answers that question
 - Launch a job, verify that variable gets used, verify that it is redacted
 - Go delete the JT survey, verify that the variable is still encrypted on the node

@simfarm this might help you to see those steps to reproduce.

Also, tangentially related...

We did have a few serializer bugs. This pushes more logic to the base LaunchConfig serializer, and fixes some things.

Previously, we did allow people to move nodes from one workflow to another. This feels very wrong to me, because nodes should be tied to their parent like hosts are to their inventory. Creating parity between those two is why the migration here is needed, but it fixes the bugs associated with null WFJTs.

@chrismeyersfsu I can understand why the `unified_job_template` necessarily had to be non-null. However, I was scratching my head trying to figure out why the WFJT field would need to be non-null. The existing field parameters [go back pretty far](https://github.com/ansible/awx/commit/39ac2c047b21da97c69f20f0eb36028d92fd93ad), so my conclusion is that it was never desirable to have this field nullable. By my testing, this is working pretty good so far.
